rfcs/0001-messages.md

```diff
@@ -0,0 +1,414 @@
+
+```rust
+/// The data required to serialize `frost::SharePackage`.
+///
+/// The dealer sends this message to each signer for this round.
+/// With this, the signer should be able to build a `SharePackage` and use
+/// the `sign()` function.
+///
+/// Note: `frost::SharePackage.public` can be calculated from `secret_share`.
+struct messages::SharePackage {
+    /// The public signing key that represents the entire group:
+    /// `frost::SharePackage.group_public`.
+    group_public: VerificationKey<SpendAuth>,
+    /// This participant's secret key share: `frost::SharePackage.share.value`.
+    secret_share: frost::Scalar,
+    /// Commitment for the signer as a single jubjub::AffinePoint.
+    /// A set of commitments to the coefficients (which themselves are scalars)
+    /// for a secret polynomial _f_: `frost::SharePackage.share.commitment`
+    share_commitment: Vec<jubjub::AffinePoint>,
+}
+
+/// The data required to serialize `frost::SigningCommitments`.
+///
+/// Each signer must send this message to the aggregator.
+/// A signing commitment from the first round of the signing protocol.
+struct messages::SigningCommitments {
+    /// The hiding point: `frost::SigningCommitments.hiding`
+    hiding: jubjub::AffinePoint,
+    /// The binding point: `frost::SigningCommitments.binding`
+    binding: jubjub::AffinePoint,
+}
+
+/// The data required to serialize `frost::SigningPackage`.
+///
+/// The aggregator decides what message is going to be signed and
+/// sends it to each signer with all the commitments collected.
+struct messages::SigningPackage {
+    /// The collected commitments for each signer as a hashmap of
+    /// unique participant identifiers: `frost::SigningPackage.signing_commitments`
+    ///
+    /// Signing packages that contain duplicate or missing `ParticipantId`s are invalid.
+    signing_commitments: HashMap<ParticipantId, SigningCommitments>,
+    /// The message to be signed: `frost::SigningPackage.message`.
+    ///
+    /// Each signer should perform protocol-specific verification on the message.
+    message: Vec<u8>,
+}
+
+/// The data required to serialize `frost::SignatureShare`.
+///
+/// Each signer sends their signatures to the aggregator who is going to collect them
+/// and generate a final spend signature.
+struct messages::SignatureShare {
+     /// This participant's signature over the message: `frost::SignatureShare.signature`
+    signature: frost::Scalar,
+}
+
+/// The data required to serialize a successful output from `frost::aggregate()`.
+///
+/// The final signature is broadcasted by the aggregator to all signers.
+struct messages::AggregateSignature {
+    /// The aggregated group commitment: `Signature<SpendAuth>.r_bytes` returned by `frost::aggregate`
+    group_commitment: jubjub::AffinePoint,
+    /// A plain Schnorr signature created by summing all the signature shares:
+    /// `Signature<SpendAuth>.s_bytes` returned by `frost::aggregate`
+    schnorr_signature: frost::Scalar,
```

Comment:
```suggestion
    schnorr_signature: jubjub::Signature
```

Reply:
I can't find `jubjub::Signature`. Are you sure is that what you meant ?

Reply:
I added a new type `frost::FrostSignature` for this.